src/schemathesis/protocols.py

```diff
@@ -0,0 +1,141 @@
+"""Protocols that describe how concrete implementations should behave."""
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, TypeVar
+
+import requests
+from hypothesis.strategies import SearchStrategy
+from typing_extensions import Protocol
+
+from .hooks import HookDispatcher
+from .stateful import StatefulTest
+from .types import Body, Cookies, FormData, Headers, PathParameters, Query
+from .utils import GenericResponse, WSGIResponse
+
+
+class CaseProtocol(Protocol):
```

Comment:
This actually should be named "Case" and the currently named class "Case" should be "OpenAPICase". Similar to "EndpointProtocol"